feature(install): basic install extension

imsnif opened the pull request:
This provides a skeleton for the `bit install` command to work as a harmony extension.

Right now this command is not connected to anything, because in order to connect it we must resolve the conflict with the legacy install command. I plan on working on this next, eventually bringing it to a state where all tests pass.

Once this is done, we can add it as a dependency to the bit extension.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/teambit/bit/2317)
<!-- Reviewable:end -->

GiladShoham commented:
You shouldn't catch inside the render and return valid react. You should throw a known exception inherent from paper error

imsnif commented:
> You shouldn't catch inside the render and return valid react. You should throw a known exception inherent from paper error

Hey, I changed it to throw a paper error that I added: https://github.com/teambit/bit/pull/2317/files#diff-e8f8d4c23979f2590e16a1bc770f3331R26. I tested this, and when this happens, I get:
```
Failed to install: aaaaa!!!
(node:1378888) UnhandledPromiseRejectionWarning: TypeError: command.handleError is not a function
    at /home/aram/code/bit/bit/dist/cli/command-registry.js:289:78
(node:1378888) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1378888) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```
I'm guessing this is a separate bug that we (maybe) know about? Should I not be throwing the error but just rendering it, or?

GiladShoham commented:
> > You shouldn't catch inside the render and return valid react. You should throw a known exception inherent from paper error
> 
> Hey, I changed it to throw a paper error that I added: https://github.com/teambit/bit/pull/2317/files#diff-e8f8d4c23979f2590e16a1bc770f3331R26. I tested this, and when this happens, I get:
> 
> ```
> Failed to install: aaaaa!!!
> (node:1378888) UnhandledPromiseRejectionWarning: TypeError: command.handleError is not a function
>     at /home/aram/code/bit/bit/dist/cli/command-registry.js:289:78
> (node:1378888) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
> (node:1378888) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
> ```
> 
> I'm guessing this is a separate bug that we (maybe) know about? Should I not be throwing the error but just rendering it, or?

You should throw it. it's a different issue we should fix.
your paper error should have a render function as well, which render the error output nicely

GiladShoham commented:
Your error looks fine.
Just move it into the install extension instead of the paper extension. 
It's install specific not paper.
